host-components/src/keep-manager-tester.rs

```diff
@@ -0,0 +1,172 @@
+    io::stdin()
+        .read_line(&mut user_input)
+        .expect("Failed to read line");
+
+    //NOTE - this should fail: not currently implemented
+    let builder = reqwest::blocking::Client::builder()
+        .danger_accept_invalid_certs(true)
+        .build()
+        .unwrap()
+        .post("https://localhost:3032/contracts_post/")
+        .json(&command1);
+    let res = builder.send();
+    println!("{:#?}", res);
+
+    println!("");
+    println!("Press <Enter> to create a Keep");
+
+    io::stdin()
+        .read_line(&mut user_input)
+        .expect("Failed to read line");
+
+    //construct a couple of keeps with command1
+    let res1: reqwest::blocking::Response = reqwest::blocking::Client::builder()
+        .danger_accept_invalid_certs(true)
+        .build()
+        .unwrap()
+        .post("https://localhost:3030/keeps_post/")
+        .json(&command2)
+        .send()
+        .expect("Possible issues");
+
+    let keeploader1: KeepLoader = res1.json().expect("Possible issues");
+    println!("Keep created with kuuid = {}", keeploader1.kuuid);
+
+    println!("");
+    println!("Press <Enter> to create another Keep");
+
+    io::stdin()
+        .read_line(&mut user_input)
+        .expect("Failed to read line");
+
+    let res2: reqwest::blocking::Response = reqwest::blocking::Client::builder()
+        .danger_accept_invalid_certs(true)
+        .build()
+        .unwrap()
+        .post("https://localhost:3030/keeps_post/")
+        .json(&command2)
+        .send()
+        .expect("Possible issues");
+    //let content2 = res2.text();
+    let keeploader2: KeepLoader = res2.json().expect("Possible issues");
+    println!("Keep created with kuuid = {}", keeploader2.kuuid);
+
+    println!("");
+    println!("Press <Enter> to list keeps.  This may include more than the Keeps you just");
+    println!("created if the keep-manager is long-lived");
+    println!("");
+
+    io::stdin()
+        .read_line(&mut user_input)
+        .expect("Failed to read line");
+
+    //list keeps
+    let res3 = reqwest::blocking::Client::builder()
+        .danger_accept_invalid_certs(true)
+        .build()
+        .unwrap()
+        .post("https://localhost:3030/keeps_post/")
+        .json(&command3)
+        .send()
+        .expect("Possible issues");
+    let keeploadervec: KeepLoaderVec = res3.json().expect("Possible issues");
+    //TODO - output
+    for keeploader in &keeploadervec.klvec {
+        println!("kuuid {}", keeploader.kuuid);
+    }
+
+    let number_of_kls = &keeploadervec.klvec.len();
+    println!("We have {} Keep-loaders", number_of_kls);
+    println!("");
+
+    println!("Press <Enter> to start the most recently created Keep");
+
+    io::stdin()
+        .read_line(&mut user_input)
+        .expect("Failed to read line");
+
+    println!(
+        "About to send start-keep command for kuuid {}, to listen on port {}",
+        &keeploadervec.klvec[number_of_kls - 1].kuuid.to_string(),
```

Comment:
`keeploadervec.klvec.last().unwrap()` instead of `keeploadervec.klvec[number_of_kls - 1]` and of course instead of `unwrap()` handling of the failure case. Imagine `number_of_kls == 0`

Reply:
Fixed this.  Also added explicit check for zero length.